src/skin/pixmapsource.cpp

```diff
@@ -48,6 +50,31 @@ void PixmapSource::setSVG(const QByteArray& content) {
     m_eType = SVG;
 }
 
+QPixmap PixmapSource::toPixmap(const QSize& size) {
```

Comment:
Ah, I think I understand how to use Paintable::draw() for this. Create a QPainter in WSpinny::resizeEvent() that paints onto the QPixmap members of WSpinny instead of onto the widget like other parts of the skin code.